airflow/operators/bash_operator.py

```diff
@@ -87,14 +88,14 @@ def execute(self, context):
 
                 self.sp = sp
 
-                logging.info("Output:")
-                line = ''
+                output = StringIO()
                 for line in iter(sp.stdout.readline, b''):
-                    line = line.decode(self.output_encoding).strip()
-                    logging.info(line)
+                    output.write(line)
+                output = output.getvalue()
```

Comment:
@artwr, current bash_operator dont have the memory issue since it completely ignore the whole output. but i agree with you on the case when output is too large, so I can add option to first write it into temp file based on user request which will give user a choice on whether he/she is okay with writing output in temp file on server ( which is not secure)